### PR TITLE
add jao converter

### PR DESCRIPTION
**Pull request by SteffenMeinecke**
- [ADDED] converter for European EHV grid data from JAO, the "Single Allocation Platform (SAP) for all European Transmission System Operators (TSOs) that operate in accordance to EU legislation"

The added code converts the European (Core) EHV grid data provided by JAO (Joint Allocation Office), the "Single Allocation Platform (SAP) for all European Transmission System Operators (TSOs) that operate in accordance to EU legislation".

**Data Sources and Availability:**
The data are available at the website `JAO Static Grid Model <https://www.jao.eu/static-grid-model>`_ (November 2024). There, a map is provided to get an fine overview of the geographical extent and the scope of the data. These inlcude information about European (Core) lines, tielines, and transformers.

### CODE REVIEW

File: pandapower/converter/jao/from_jao.py [hunk 1]
```diff
@@ -0,0 +1,987 @@
...
+        cols.loc[cols[1] == "Comment", 0] = None
+        cols.loc[cols[0].str.startswith("Unnamed:").astype(bool), 0] = None
+        cols.loc[cols[1] == "Length_(km)", 0] = "Electrical Parameters"  # might be wrong in
+        # Tielines otherwise
+        data[key].columns = pd.MultiIndex.from_arrays(cols.values.T)
+
+        # --- correct comma separation and cast to floats
+        data[key][("Maximum Current Imax (A)", "Fixed")] = \
+            data[key][("Maximum Current Imax (A)", "Fixed")].replace("\xa0", 999e3).replace(
+                "-", 999e3).replace(" ", 999e3)
```
Comment (jkisse) [inline pandapower/converter/jao/from_jao.py:166]: I think overwriting of N/A imax should not be done, or at least be optional and with a warning.
Max. I DLR is often provided for these lines and could serve as a substitute (sth. like a flag `apply_DLRmax_for_missing_max_i`)

File: pandapower/converter/jao/from_jao.py [hunk 2]
```diff
@@ -0,0 +1,987 @@
...
+    # --- Transformer data --------------------------------
+    key = "Transformers"
+
+    # --- fix Locations
+    loc_name = ("Location", "Full Name")
+    data[key].loc[:, loc_name] = data[key].loc[:, loc_name].str.strip().str.replace(
+        "PSTMIKULOWA", "PST MIKULOWA").str.replace(
+        "Chelm", "CHELM").str.replace(
+        "OLSZTYN-MATK", "OLSZTYN-MATKI").str.replace(
+        "STANISLAWOW", "Stanislawow").str.replace("VIERRADEN", "Vierraden")
```
Comment (jkisse) [inline pandapower/converter/jao/from_jao.py:188]: For maintainability, I'd prefer a less chained approach, like
```
# old name -> new name
rename_locnames = {"PSTMIKULOWA": "PST MIKULOWA",
                   "Chelm": "CHELM", 
                   "OLSZTYN-MATK": "OLSZTYN-MATKI", 
                   "STANISLAWOW": "Stanislawow", 
                   "VIERRADEN": "Vierraden"} 
data[key].loc[:, loc_name] = data[key].loc[:, loc_name].str.strip().replace(rename_locnames )
```

File: pandapower/converter/jao/from_jao.py [hunk 3]
```diff
@@ -0,0 +1,987 @@
...
+            net,
+            bus_idx.loc[list(tuple(zip(data[key].loc[:, ("Substation_1", "Full_name")].values,
+                        vn_kvs)))].values,
+            bus_idx.loc[list(tuple(zip(data[key].loc[:, ("Substation_2", "Full_name")].values,
+                        vn_kvs)))].values,
+            length_km,
+            data[key][("Electrical Parameters", "Resistance_R(Ω)")].values / length_km,
+            data[key][("Electrical Parameters", "Reactance_X(Ω)")].values / length_km,
+            data[key][("Electrical Parameters", "Susceptance_B(μS)")].values / length_km,
+            data[key][("Maximum Current Imax (A)", "Fixed")].fillna(999000).values / 1e3,
```
Comment (jkisse) [inline pandapower/converter/jao/from_jao.py:325]: as mentioned above, dummy max_I should not be set silently and without explicit permission from the user

**Comment by vogt31337** [comment]
@SteffenMeinecke could you provide a small description?

**Comment by SteffenMeinecke** [comment]
> @SteffenMeinecke could you provide a small description?

done

**Comment by KS-HTK** [comment]
How did this get merged with 0 Approval reviews?

**Pull request merged**